xbmc/video/ContextMenus.cpp

```diff
@@ -58,6 +58,8 @@ bool CMarkWatched::IsVisible(const CFileItem& item) const
   {
     if (item.HasVideoInfoTag())
       return item.IsVideoDb();
+    else if (item.GetProperty("IsVideoFolder").asBoolean())
```

Comment:
If this can only be a recording folder or a video item, why not change the if-else to check video info tag, is recording, not parent folder and else return false.